src/useEvent.ts

```diff
@@ -1,12 +1,13 @@
-import { useLayoutEffect, useRef } from "react";
+// @ts-ignore
```

Comment:
1. Let's split the imports so we can ignore just the `useInsertionEffect` line
2. Let's use `// @ts-expect-error` so it'll be cleaned up eventually

Reply:
I had this solution first, but saw that the build output in commonjs was generating 2 react imports aswell don't know if this has any unwanted implications. 

If you think it doesn't i will change it happily. 

Maybe this is also a good chance to include an esm output.

Reply:
I made a small change to the imports, so that we're just importing once.  Thanks for making all the updates!